### ISSUE TITLE
Multiple specific or general msName elements?

### ISSUE THREAD

**Issue by djbpitt** [djbpitt/repertorium#14]
Some mss have multiple specific or general `<msName >`elements. We use a single `<msName>` as the name of the ms in listings (such as in the browsing interface), which means that although multiple names of the same type are allowed and can be seen in the full description, only one is rendered in short title listings. The order of names for a ms where there is more than one does not seem to be specified or constrained, though, so in situation where there is more than one, the one that winds up getting rendered in short title listings appears to be arbitrary.

Neither the *Scripta* documentation of `<msName>` (pp. 22–23) nor the [HTML addendum](http://repertorium.obdurodon.org/encoding-policies-update.xhtml#msName) says anything about how to order multiple specific or generic names. Shouldn't we document and enforce a consistent policy?

**Comment by atoboy** [comment]
See for some visualizations here: https://www.slav.uni-sofia.bg/zograf/pages/show.html?document=Zogr_0004.xml
Here &lt;msName type="specific"&gt;...&lt;/msName&gt; is rendered as Конкретно наименование: .
We can ordered this kind of information as follows:
1. Generic (general) name
2. Specific name
3. Individual name

**Comment by djbpitt** [comment]
@atoboy Thank you, Andrej, for the quick response. I regret, though, that my original question was not as clear as I had hoped it would be, since I was asking not about what to render in a codicological description (of the sort to which you link, above), but in a list of manuscripts like http://repertorium.obdurodon.org/browse-checkbox.php. For that view we give just a single manuscript "name", and our protocol is:

1. If there is an individual name, use that.
1. If there isn't an individual name but there is a specific name, use that (or, if there is more than one, use the first one).
1. If there isn't an individual or a specific name, use the first general name.

There can be more than one name of each type, and we can render them all _in the codicological description_, ordered however we like. _In the short listing,_ though, we want to give just a single name. 

The issue I was asking about is that we don't seem to enforce an order for multiple specific names or multiple general names. This means that if we select whichever one is first for a short listing, we wind up not making that choice in a consistent way. For example, AS9D15MP.xml has no individual or specific names and it has two general ones: first "miscellany" and then "panegyricon". I don't see anything in our documentation about whether, in that situation, "miscellany" should come first or "panegyricon" should come first, and since in the short list we use only one name, the choice seems to be important. My intuition is that we should use the narrower term (in this case, "panegyricon") because it is more informative than the broader one (in this case, "miscellany"). If you agree, though, we should document that (I can do it in our HTML supplement) and we should edit the existing descriptions to put multiple entries of the same type (that is, multiple specific or multiple general names) in order from narrower to broader. If we do that, the code that chooses the first one for the short lists will select the one we consider more informative.

I'm asking this now because I've recently resumed work on transforming the Repertorium materials into a self-contained eXist-db app so that I can remove the current dependency on PHP middleware. This will make it easier to maintain the Repertorium materials. I'm also improving the underlying XQuery and XSLT and using new indexing features that were recently added to eXist-db (specifically facets and fields).

Do you concur with my suggestion that we should encode multiple specific or multiple general names in order from narrower to broader, so that in situations where we need to choose just the first one, we will consistently choose the most informative one?

**Comment by atoboy** [comment]
I would suggest that we should have just one general name. In the example of AS9D15MP.xml having "miscellany" and "panegyricon" simultaneously is a mistake. We should have just "panegyricon" in this case. As you know most of the medieval MSS could be considered as "miscellany" of some sort, so this name is not informative. 
I will propose to Anissava and Dilyana that we go through all descriptions and edit all instances of general and specific names of the MSS. For me, there should be just one general and one specific name for each manuscript.

**Comment by djbpitt** [comment]
@atoboy Agreed, and thank you! Please pull the master branch from the GitHub repo and edit the versions of the files located there at repertorium/app/data/mss. I have to teach in a few minutes, but after class today I'll update the Schematron to raise an error if there is not exactly one general name and one specific name for each manuscript.

I've made a lot of progress on the app interface and I think it's an improvement over the current published version, but there's also still a lot to do before it's ready for serious review. I'll keep you all informed, and once all the pieces are working I'll ask you all for suggestions and other feedback.

**Comment by djbpitt** [comment]
@atoboy I have modified the Schematron to require exactly one general and exactly one specific name for each manuscript. Most manuscripts are invalid in this respect, usually because they have multiple specific names, but sometimes for other reasons (no specific name, multiple general names, no general name). If you pull and validate all of the manuscripts, \<oXygen/\> will find the offending markup.
